test/Test/BaseDAO.hs

```diff
@@ -0,0 +1,166 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.BaseDAO
+  ( test_BaseDAO
+  ) where
+
+import Universum
+
+import qualified Data.Map.Strict as Map
+import Lorentz
+import Lorentz.Test
+import Morley.Nettest
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase)
+import Util.Named
+
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import qualified Indigo.Contracts.SampleFA2 as SampleFA2 -- TODO: Replace SampleFA2 with BaseDAO
+
+-- TODO: Replace SampleFA2 with BaseDAO
+baseDAOStorage
+  :: Map (Address, FA2.TokenId) Natural
+  -> SampleFA2.Operators -> SampleFA2.Storage
+baseDAOStorage bal operators =
+  SampleFA2.mkStorage daoToken bal operators
+  where
+    daoToken :: [ FA2.TokenMetadata ]
+    daoToken =
+      [ FA2.TokenMetadata
+          { token_id = 0
+          , symbol = [mt|unfrozen_token|]
+          , name = [mt|Unfrozen Token|]
+          , decimals = 8
+          , extras = Map.empty
+          }
+      , FA2.TokenMetadata
+          { token_id = 1
+          , symbol = [mt|frozen_token|]
+          , name = [mt|Frozen Token|]
+          , decimals = 8
+          , extras = Map.empty
+          }
+      ]
+
+-- TODO: Replace SampleFA2 with BaseDAO
+baseDAOContract :: FA2.PermissionsDescriptor -> Contract FA2.Parameter SampleFA2.Storage
+baseDAOContract p =
+  SampleFA2.fa2Contract p
+
+originateBaseDAO
+  :: MonadNettest caps base m => FA2.PermissionsDescriptor
+  -> m ((Address, Address), (Address, Address), TAddress FA2.Parameter)
+originateBaseDAO p = do
+  owner1  :: Address <- newAddress "owner1"
+  operator1  :: Address <- newAddress "operator1"
+  owner2  :: Address <- newAddress "owner2"
+  operator2  :: Address <- newAddress "operator2"
+
+  let bal = Map.fromList
+        [ ((owner1, 0), 100)
+        , ((owner2, 0), 100)
+        ]
+  let (operators :: SampleFA2.Operators) =
+        BigMap $ Map.fromList
+          [ ((owner1, operator1), ())
+          , ((owner2, operator2), ())
+          ]
+  let
+    originateData = OriginateData
+      { odFrom = AddressResolved genesisAddress
+      , odName = "BaseDAO"
+      , odBalance = toMutez 0
+      , odStorage = baseDAOStorage bal operators
+      , odContract = baseDAOContract p
+      }
+  dao <- originate originateData
+  pure ((owner1, operator1), (owner2, operator2), dao)
+
+test_BaseDAO :: TestTree
+test_BaseDAO = testGroup "tests to check BaseDAO contract functionality"
+  [ testCase "Allows zero transfer from non-existent accounts" $
+      nettestTestExpectation zeroTransferScenario
+  , testCase "Is denied if only owner transfer is allowed" $
```

Comment:
We've decided to use the default permissions descriptor that has `Owner_or_operator_transfer`, so operator transfer is allowed and this test looks redundant.